BrainPortal/app/controllers/userfiles_controller.rb

```diff
@@ -188,6 +188,29 @@ def create_parent_child #:nodoc:
     redirect_to :action => :index
   end
 
+  # Use the rest of the route as a path into a file collection.
+  # [:file_path] a path relative to the file collection's route directory.
+  #
+  # GET /userfiles/1/file_collection_content/path/to/file/in/file_collection
+  def file_collection_content
+    @userfile = Userfile.find_accessible_by_user(params[:id], current_user, :access_requested => :read)
+    if @userfile.nil?
+      raise ActiveRecord::RecordNotFound
```

Comment:
Add a nice message to the exception? `ActiveRecord::RecordNotFound.new('Message')`

Reply:
oops!